### DISCUSSION TITLE
Block config > image url should be a component not an url

### DISCUSSION THREAD

**Discussion by ciokan** [payloadcms/payload#1092] [Roadmap]
On the [block config](https://payloadcms.com/docs/fields/blocks#block-configs) you have the option of setting up a image url, representative of that block. I don't think that's ideal because it forces us to hook up with express to serve files that render poorly (for an svg I had to set a height for it to even show up) or brings complications when the theme is on dark vs light, when you want different colors etc.

I think that the best solution would be a component. You guys do it with the UI field: https://payloadcms.com/docs/fields/ui This makes tools like [heroicons](https://heroicons.com/) really accessible, themable, configurable from within the client.

**Comment by jmikrut** [comment]
@ciokan this is a great idea and would be relatively trivial to PR. We would happily merge this. 

Any interest in taking a crack? Otherwise, I'm sure we can get to it! Just not sure when. But it'll be a quick one and it's a great idea.

**Reply by ciokan** [reply]
> @ciokan this is a great idea and would be relatively trivial to PR. We would happily merge this.
> 
> Any interest in taking a crack? Otherwise, I'm sure we can get to it! Just not sure when. But it'll be a quick one and it's a great idea.

Yes. I'll have a go at it this weekend I hope.

**Reply by ciokan** [reply]
Should it replace image* fields or the component as an alternative? I think it should replace since the developer will have full control over the component and could add an image if he/she choses to so I don't see any point in keeping those 2 fields unless for backwards compatibility.

**Reply by jmikrut** [reply]
I would say in order to avoid a breaking change, this should be an _alternative_ instead of a replacement. Yep. Backwards compatibility concerns and ease-of-use in case someone does indeed want to just pass an image. Great question!

**Reply by ktwebmasters99** [reply]
Great way to think

**Comment by MarkAtOmniux** [comment]
@jmikrut  @ciokan  has any work been started on this? If not, i'd be happy to take a crack at it :)
